ddtrace/tracer/sampler.go

```diff
@@ -158,49 +158,115 @@ func (ps *prioritySampler) apply(spn *span) {
 	spn.SetTag(keySamplingPriorityRate, rate)
 }
 
-// rulesSampler allows a user-defined list of rules to apply to spans.
-// These rules can match based on the span's Service, Name or both.
-// When making a sampling decision, the rules are checked in order until
-// a match is found.
-// If a match is found, the rate from that rule is used.
-// If no match is found, and the DD_TRACE_SAMPLE_RATE environment variable
-// was set to a valid rate, that value is used.
-// Otherwise, the rules sampler didn't apply to the span, and the decision
-// is passed to the priority sampler.
-//
-// The rate is used to determine if the span should be sampled, but an upper
-// limit can be defined using the DD_TRACE_RATE_LIMIT environment variable.
-// Its value is the number of spans to sample per second.
-// Spans that matched the rules but exceeded the rate limit are not sampled.
+// rulesSampler holds instances of trace sampler and single span sampler, that are configured with the given set of rules.
+// traceRulesSampler samples trace spans based on a user-defined set of rules and might impact sampling decision of the trace.
+// singleSpanRulesSampler samples individual spans based on a separate user-defined set of rules and
+// cannot impact the trace sampling decision.
 type rulesSampler struct {
-	rules      []SamplingRule // the rules to match spans with
-	globalRate float64        // a rate to apply when no rules match a span
-	limiter    *rateLimiter   // used to limit the volume of spans sampled
+	traceRulesSampler      *traceRulesSampler
+	singleSpanRulesSampler *singleSpanRulesSampler
```

Comment:
```suggestion
	traces *traceRulesSampler
	spans  *spanRulesSampler
```
Would this not be easier to work with?